File: characteristics/service-name.js
let util = require('util')
let bleno = require('@stoprocent/bleno')
let UUID = require('../sugar-uuid')

let BlenoCharacteristic = bleno.Characteristic
let BlenoDescriptor = bleno.Descriptor

module.exports = new BlenoCharacteristic({
  uuid: UUID.SERVICE_NAME,
  properties: ['read'],
  value: Buffer.from('PiSugar BLE Wifi Config'),
  descriptors: [
    new BlenoDescriptor({
      uuid: '2001',
      value: 'PiSugar BLE Wifi Config'
    })
  ]
})

File: rust/src/ble/service_name.rs
use bluer::gatt::local::{Characteristic, CharacteristicRead, CharacteristicReadRequest};
use futures::FutureExt;

use crate::uuid as suuid;

/// Build the Service Name characteristic (read-only, static).
pub fn build() -> Characteristic {
    Characteristic {
        uuid: suuid::parse_uuid(suuid::SERVICE_NAME),
        read: Some(CharacteristicRead {
            read: true,
            fun: Box::new(move |_req: CharacteristicReadRequest| {
                async move {
                    log::debug!("ServiceName read request");
                    Ok(b"PiSugar BLE Wifi Config".to_vec())
                }
                .boxed()
            }),
            ..Default::default()
        }),
        ..Default::default()
    }
}
